uniffi_bindgen/src/interface/attributes.rs

```diff
@@ -27,6 +27,7 @@ pub(super) enum Attribute {
     ByRef,
     Enum,
     Error,
+    Interface,
```

Comment:
I don't feel too strongly about this, but another option might be to have the UDL instead do, say ```[ExternalInterface="uniffi-one"]``` - this would mean `External` and `ExternalInterface` end up being mutually exclusive but that should be an easy check to make

This would also tie in with comments below re `bool` vs a new enum - eg, you could have
```
enum ExternalImpl {
  Interface, // better names though? Best to avoid 'Object' though :)
  Other, // "TheRest"? :)
}
```

And `Attribute::External` could just get one of these (along with the existing String). When parsing the attributes, both the literals `"External"` and `"ExternalInterface"` create an `Attribute::External`, just with a different discriminator.

You could then also use this new enum in `Type::External`

That would avoid a new attribute completely and avoid the new `is_interface()` function (but you probably would need something like a new `get_external_imp(&self) -> Option<ExternalImpl>` or something for finder.rs to know how to create the `Type::External`.

If we thought this would ever be needed for other types, another possibility might be, say, ```[External="uniffi-one", ExternalType=interface]``` - you'd still need the new Attribute, but it could probably avoid leaking much from that module (ie, the "public" API would still be to just get the enum without knowing exactly what attributes were used to specify it - but that seems a little like yagni.